src/general.md

```diff
@@ -129,6 +129,28 @@ In the UI, the language should be local. For the UK, Singapore, and HK, it's Bri
    isDiscountAdjustmentDisabled;
    ```
 
+1. Avoid third-party names of softwares or services.
+
+   ```typescript
+   // bad
+
+   <HotJar company={company} />
+
+   // good
+
+   <Nps company={company} />
+   ```
+
+   ```typescript
+   // bad
+
+   company.xeroData;
```

Comment:
not sure about it

Reply:
To be honest, I see two cases. The name of third-party:
1. as a link to an entity in this service. It might be okay.
e.g. xeroId, hubstopId.
2. as a concrete realisation of something big.
e.g. HotJar component or service instead of Nps.


About bullet 1
xeroId is a strange name, better is xeroDocumentId. externalDocumentId + externalSystem or just externalDocumentId + knowledge in code or comment about the currently used system.
At the same time, we have cases when we have several services as providers. e.g. Jamal with many providers or systemId in Invoker microservices. I'm sure, that we can skip the first step with a straightforward solution.
Another good example, is Sendbird, now we can find many-many functions with this name.

Reply:
Sounds like we can cut like avoid the third-party name in functions, components and service files. Only for SDK files.
Don't describe table columns and model fields.

Reply:
the main argument why we call it `<name>Data` is because we don't control it in terms of the shape. So we do it when the type of the object is just `Object` or some shape directly imported from external SDK or generally when we just leave it as is. And here the name give you this understanding that it's not some owned by us accounting data, it's just Xero data.

The second argument is that from time to time we switch external services it help a lot when they have their unique names.

I fully agree that ideally you would want to see them more abstract, but from the practical point of view still it seems like giving unique names is better while not being ideal.

Reply:
Updated the wording. Please check one more time. If no, then I'll clone this PR.